### PR TITLE
Fix GCC errors about OE_RAISE jumping variable initialization

### PR DESCRIPTION
**Pull request by wintersteiger**
Fixes GCC errors about the `goto` in `OE_RAISE` jumping variable initializations (`-Werror=jump-misses-init`) in EEID code; no semantic changes.

Signed-off-by: Christoph M. Wintersteiger <cwinter@microsoft.com>

### CODE REVIEW

File: enclave/crypto/sha.c [hunk 1]
```diff
@@ -121,7 +121,7 @@ oe_result_t oe_sha256_restore(
 
     uint64_t NB = ((((uint64_t)num_hashed[1]) << 32) + num_hashed[0]) / 8;
     impl->ctx.total[0] = NB & 0xFFFFFFFF;
-    impl->ctx.total[1] = (NB >> 32) & 0xFFFFFFFF;
+    impl->ctx.total[1] = ((uint32_t)(NB >> 32)) & 0xFFFFFFFF;
```
Comment (CodeMonkeyLeet) [inline enclave/crypto/sha.c:124]: For my understanding, why is this necessary, and also, why is it not necessary for the line above this? In either case, if we're cast truncating to 32-bits, it seems the extra bit masking operation is unnecessary.

File: common/sgx/eeid.c [hunk 2]
```diff
@@ -132,13 +132,12 @@ static oe_result_t deserialize_element(
 oe_result_t oe_serialize_eeid(const oe_eeid_t* eeid, char* buf, size_t buf_size)
 {
     oe_result_t result = OE_UNEXPECTED;
+    char** position = &buf;
```
Comment (CodeMonkeyLeet) [inline common/sgx/eeid.c:135]: This is an unsafe pointer dereference because the null check happens after this line now.

File: common/sgx/eeid.c [hunk 3]
```diff
@@ -209,13 +208,12 @@ oe_result_t oe_deserialize_eeid(
     oe_eeid_t* eeid)
 {
     oe_result_t result = OE_UNEXPECTED;
+    const char** position = &buf;
```
Comment (CodeMonkeyLeet) [inline common/sgx/eeid.c:211]: Similarly, unsafe pointer dereference.

File: common/sgx/eeid.c [hunk 4]
```diff
@@ -794,21 +800,20 @@ oe_result_t oe_eeid_evidence_ntoh(
     oe_eeid_evidence_t* evidence)
 {
     oe_result_t result = OE_UNEXPECTED;
+    const uint8_t* position = buffer;
+    size_t remaining = buffer_size, data_size;
```
Comment (CodeMonkeyLeet) [inline common/sgx/eeid.c:804]: Style: please split variable declarations with initializations apart from those without for readability.

```suggestion
    size_t remaining = buffer_size;
    size_t data_size = 0;
```

File: common/sgx/eeid.c [hunk 5]
```diff
@@ -765,21 +772,20 @@ oe_result_t oe_eeid_evidence_hton(
     size_t buffer_size)
 {
     oe_result_t result = OE_UNEXPECTED;
+    uint8_t* position = buffer;
+    size_t remaining = buffer_size, data_size;
```
Comment (CodeMonkeyLeet) [inline common/sgx/eeid.c:776]:
```suggestion
    size_t remaining = buffer_size;
    size_t data_size = 0;
```

File: common/sgx/eeid.c [hunk 6]
```diff
@@ -390,6 +388,11 @@ static oe_result_t verify_base_image_signature(const sgx_sigstruct_t* sigstruct)
     oe_result_t result = OE_UNEXPECTED;
     unsigned char buf[sizeof(sgx_sigstruct_t)];
     size_t n = 0;
+#ifndef OE_BUILD_ENCLAVE
```
Comment (CodeMonkeyLeet) [inline common/sgx/eeid.c:391]: Consider splitting this function into dedicated subfunctions for mbedtls and OpenSSL handling; the interleaved `#ifdefs` make the code increasingly hard to follow.

Side note: it looks like the mbedtls section of the code should have the same compiler issue, but may not be flagged because it is not being built with GCC in your current environment?

File: enclave/crypto/sha.c (hunk 1 above)
Comment (wintersteiger) [reply enclave/crypto/sha.c:124]: It's necessary because GCC complains (clang doesn't). I think the issue is the automatic promotion of `(NB >> 32)` to a signed type, but I didn't investigate it any further.

File: common/sgx/eeid.c (hunk 2 above)
Comment (wintersteiger) [reply common/sgx/eeid.c:135]: There's no dereference here, it just takes the address of the argument, which should always be safe, even if `buf` is `NULL`.

File: common/sgx/eeid.c (hunk 3 above)
Comment (wintersteiger) [reply common/sgx/eeid.c:211]: (see above)

File: common/sgx/eeid.c (hunk 6 above)
Comment (wintersteiger) [reply common/sgx/eeid.c:391]: This fixes both mbedTLS and OpenSSL, but  yeah, I think the time has come to split that up :-)

Comment (wintersteiger) [reply common/sgx/eeid.c]: (done)

**Comment by wintersteiger** [comment]
bors try

**Comment by CodeMonkeyLeet** [comment]
bors try

**Comment by wintersteiger** [comment]
The bors build was successful, but GitHub says it's still waiting for the status to be reported. Is there something wrong with the infrastructure?

**Comment by anakrish** [comment]
bors try

**Comment by wintersteiger** [comment]
bors try

**Comment by wintersteiger** [comment]
Bors was successful a second time, but the status is still "expected". Is there something wrong with bors or github?

**Comment by radhikaj** [comment]
> Bors was successful a second time, but the status is still "expected". Is there something wrong with bors or github?

@BRMcLaren  @rs-- Do you have insights here?

**Comment by radhikaj** [comment]
bors try

**Comment by BRMcLaren** [comment]
We had unexpected down time and our monitoring didn't catch it over the weekend as it was primarily web server related versus functionality. Lots of unanswered web hooks.

Sorry about that.

**Comment by wintersteiger** [comment]
I had to manually navigate to Blue Ocean to see the build logs and it looks like there is some sort of problem, but I have no idea what the hudson exception means. (Also, there seems to have been a problem with mail delivery.)

```
100% tests passed, 0 tests failed out of 265
...
Caught global pipeline exception: hudson.AbortException: script returned exit code 1
[Pipeline] emailext
Sending email to: cwinter@microsoft.com
Connection error sending email, retrying once more in 10 seconds...
Connection error sending email, retrying once more in 10 seconds...
Failed after second try sending email
[Pipeline] End of Pipeline
ERROR: script returned exit code 1
Finished: FAILURE
```

**Comment by anakrish** [comment]
bors try

**Comment by wintersteiger** [comment]
The latest build reports this:

```
fatal: [localhost]: FAILED! => {"changed": false, "dest": "/tmp/shellcheck.tar.xz", "elapsed": 0, "msg": "Request failed", "response": "HTTP Error 404: Not Found", "status_code": 404, "url": "https://storage.googleapis.com/shellcheck/shellcheck-v0.4.7.linux.x86_64.tar.xz"}
```

I'll try again later.

**Comment by CodeMonkeyLeet** [comment]
@wintersteiger Should have been fixed by #3344.

bors try
